ravenframework/Driver.py

```diff
@@ -141,6 +141,10 @@ def main(checkLibraries):
 
     # call the function to load the external xml files into the input tree
     cwd = os.path.dirname(os.path.abspath(inputFile))
+    # If a user chooses to run RAVEN outside of the directory containing the
+    # xml input file, then we should should change the python working directory
+    # to the directory containing the input file to avoid confusion.
+    os.chdir(cwd)
```

Comment:
I think this change just solve part of the problem. Take the test "raven/tests/framework/test_iostep_load.xml" for example (see the following attached screenshot), the files provided by Files using relative path, which I think currently based on the WorkingDir provided by the RunInfo. I think the confusion here is we do not have a consistent way to explain and use the path. The issue #1687 suggest we use the input file folder as the base directory.  The following is my suggestion: I think the default behavior is use the input file folder as the base directory and search the files, if the files do not exist, then using the WorkingDir listed in RunInfo as the base directory, if the files are still not found, then use the running directory as the base directory. After trying search all these places, error will be raise if we can not find these files. @dylanjm @Jimmy-INL @joshua-cogliati-inl @PaulTalbot-INL @mandd please let me know what do you think.  

![image](https://user-images.githubusercontent.com/7321071/176276134-08f18f48-066c-4a30-a387-2b01213f6e4f.png)